kedify-proxy/templates/envoy-deployment.yaml

```diff
@@ -0,0 +1,105 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "kedify-proxy.fullname" . }}
+  labels:
+    {{- include "kedify-proxy.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.deployment.replicas }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "kedify-proxy.selectorLabels" . | nindent 6 }}
+  {{- with .Values.deployment.rollingUpdate }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      {{- toYaml . | nindent 6 }}
+  {{- end }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "kedify-proxy.labels" . | nindent 8 }}
+        {{- with .Values.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      containers:
+        - name: envoy
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          args:
+            - "--log-format"
+            - {"Timestamp":"%Y-%m-%dT%T.%F","ThreadId":"%t","SourceLine":"%s:%#","Level":"%l","Message":"%j"}
+            - "--config-yaml"
+            - "$(ENVOY_CONFIG)"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: ENVOY_UID
+              value: "0"
```

Comment:
@wozniakjan does that mean that we are running this as a superuser?

Reply:
changed this to 101 & also added security context for the container